File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        int answer = 0;
        for(int i = 0; i < n; i++){
            int count = 1;
            for(int j = i+1; j<n; j++){
                if(nums[i] == nums[j]){
                    count++;
                }
            }
            if(count > n/2){
                answer = nums[i];
            }
        }
        // int supercount = 0;
        // int answer = nums[0];
        // for(int i =0; i<n ; i++){
        //     int count = 0;
        //     for(int j = 0; j<n; j++){
        //         if(nums[i] == nums[j]){
        //             count++;
        //         }
        //     }
        //     if(count >= supercount){
        //             supercount = count;
        //             answer = nums[i];
        //         }
        // }
        return answer;
    }
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        if(nums.size() == 1){
            return nums[0];
        }
        sort(nums.begin(), nums.end());
        return(nums[nums.size()/2]);
        // int count = 0;
        // for(int i = 0; i < nums.size()-1; i++){
        //     if (nums[i] == nums[i+1]){
        //         count++;
        //     }
        //     if(count >= nums.size()/2){
        //         return nums[i];
        //     }
        // }
        // return -1;
    }
};